torch/_inductor/codegen/cpp.py

```diff
@@ -2146,6 +2142,8 @@ def __init__(
             tiling_factor = self.vec_isa.nelements(dtype=tiling_dtype)
         self.tiling_factor = tiling_factor
         self.tiling_idx = tiling_idx
+        self.tail_size = tail_size
+        self.num_elems = tail_size if tail_size else tiling_factor
```

Comment:
How do we handle the case when `self.tail_size` is symbolic? This seems problematic when we use `self.tail_size` as constant in the code like in `GCC unroll`?

Reply:
I checked `has_free_symbols` in `CppVecKernelChecker`, for dynamic shapes, the tail loop still uses scalar kernel.